swxsoc/util/config.py

```diff
@@ -248,3 +194,90 @@ def copy_default_config(overwrite=False):
             warn_user(message)
     else:
         shutil.copyfile(config_file, user_config_file)
+
+
+def print_config(config):
+    """
+    Print the current configuration options.
+
+    Args:
+        config (dict): The configuration data to print.
+    """
+    print("FILES USED:")
+    for file_ in _find_config_files():
+        print("  " + file_)
+
+    print("\nCONFIGURATION:")
+    for section, settings in config.items():
+        print(f"  [{section}]")
+        for option, value in settings.items():
+            print(f"  {option} = {value}")
+        print("")
+
+
+def _find_config_files():
+    """
+    Find the locations of configuration files.
+
+    Returns:
+        list: A list of paths to the configuration files.
+    """
+    config_files = []
+    config_filename = "config.yml"
+
+    # find default configuration file
+    module_dir = Path(swxsoc.__file__).parent
+    config_files.append(str(module_dir / "data" / config_filename))
+
+    # if a user configuration file exists, add that to list of files to read
+    # so that any values set there will override ones specified in the default
+    # config file
+    config_path = Path(_get_user_configdir())
+    if config_path.joinpath(config_filename).exists():
+        config_files.append(str(config_path.joinpath(config_filename)))
+
+    return config_files
+
+
+def get_and_create_download_dir():
+    """
+    Get the download directory from the configuration and create it if it doesn't exist.
+
+    Returns:
+        str: The path to the download directory.
+
+    Raises:
+        RuntimeError: If the download directory is not writable.
+    """
+    download_dir = os.environ.get("SWXSOC_CONFIGDIR")
+    if download_dir:
+        return download_dir
+
+    config = load_config()
+    download_dir = Path(config["downloads"]["download_dir"]).expanduser().resolve()
+    if not _is_writable_dir(download_dir):
+        raise RuntimeError(
+            f'Could not write to swxsoc downloads directory="{download_dir}"'
+        )
+
+    return config["downloads"]["download_dir"]
+
+
+def get_and_create_sample_dir():
+    """
+    Get the sample data directory from the configuration and create it if it doesn't exist.
+
+    Returns:
+        str: The path to the sample data directory.
+
+    Raises:
+        RuntimeError: If the sample data directory is not writable.
+    """
+    config = load_config()
+    sample_dir = Path(config["downloads"]["sample_dir"]).expanduser().resolve()
```

Comment:
What is this directory used for? I'm not sure if this is used anywhere in the code base anymore.

Reply:
I think it's a carryover from the original configuration environment, if it's not used I'll clean it up